tests/_internals/forbidden_warnings.py

```diff
@@ -62,6 +62,11 @@ def pytest_itemcollected(self, item: pytest.Item):
             # Add marker at the beginning of the markers list. In this case, it does not conflict with
             # filterwarnings markers, which are set explicitly in the test suite.
             item.add_marker(pytest.mark.filterwarnings(f"error::{fw}"), append=False)
+        item.add_marker(
```

Comment:
@potiuk 
This fix looks like a hack. It's working. However, I didn't find better way to handle this.

Reply:
As mentioned above - after looking at it - I changed my mind and I think we should at least have  a way to not issue warning (and maybe use it in tests) - so likely 3-state configuration: consistent, legacy (default - with warnings), legacy-nowarning. I foresee a number of people will have a problem with switching to consistent metrics and might defer it till airlfow 3 so finding a way to disable the warnings is a good idea (and we could set the "legacy-no-warning" setting to be default in unit tests).

Reply:
Earlier code issues warning on each metric published. Now, we have moved this to the class level. It will be issued only during the module load time. So that way, the number of warnings are less and more informative to users. Please do let me know your thoughts on this.

Reply:
I am a bit torn on that one @ferruzzi @kaxil - WDYT? Would it be ok to have a warning there during module load that the metrics should be switch to "consistent" ?

Reply:
(and BTW - I assume that the "hack" is really because this is happenig at load time, and we cannot really catch them in individual tests?)

Reply:
Yes. This is happening during airflow test setup time before the execution of actual test.

Reply:
As I said on Slack, I don't see anything that changes my previous comment from last month.

> To add to that, even if we do standardize them, why break two (Datadog and Otel) to match one (StatsD)? If we were going to do this, shouldn't we change the one that doesn't match the other two?


We get YACO (yet another config option) but we're still breaking two backends.   Without some usage numbers to see what percentage of users are going to be affected here, I don't agree with this.

Reply:
> As I said on Slack, I don't see anything that changes my previous comment from last month.
> 
> > To add to that, even if we do standardize them, why break two (Datadog and Otel) to match one (StatsD)? If we were going to do this, shouldn't we change the one that doesn't match the other two?

It's not about breaking. It's about doing the right thing. If we go with the milliseconds, then we have high precision and I am not sure with seconds. If seconds is the right way, then we will go with it. Let's hear from @potiuk / @kaxil.

> 
> We get YACO (yet another config option) but we're still breaking two backends.

This option is temporary and required during the transition and we can remove it at a later point in time.  If you have any better option in mind, then please do share it with us.

> Without some usage numbers to see what percentage of users are going to be affected here, I don't agree with this.

I do agree the consumers are impacted. I don't think we have information on the number of airflow consumers by their telemetry setup.

Reply:
I'd say - it's "right" to do, but I 'd also say - since we have Airflow 3 coming, this could be another breaking change that could be done as part of Airflow 3 implementation without adding config option. And should be accompanied with explanation on what the users should do in their metrics and dashboards - basically migration notes explaining what is going to change.

Reply:
> Hi @potiuk / @kaxil,
> I am looking for the comment on the above suggestion "Using seconds instead of milli seconds" from @ferruzzi.
> Please let us know your thoughts on this.

@kaxil Can you share your comments on this? So, that we all align on it and will make the necessary changes.

Reply:
@potiuk / @kaxil 
I am still confused about the final timer metric unit for unification. is it milli seconds or seconds? Can you please clarify it?
@ferruzzi recommends going with seconds instead of milliseconds in the earlier conversation. I am based on milliseconds. However i am good with seconds as well.

Reply:
Resolving this conversation as we aligned on the metric unit milli seconds.